book/src/specs/kimchi.md

```diff
@@ -415,6 +535,145 @@ where $w_{i, next}$ is the polynomial $w_i(\omega x)$ which points to the next r
 
 #### Chacha 
 
+There are four chacha constraint types, corresponding to the four lines in each quarter round.
+
+```
+a += b; d ^= a; d <<<= 16;
+c += d; b ^= c; b <<<= 12;
+a += b; d ^= a; d <<<= 8;
+c += d; b ^= c; b <<<= 7;
+```
+
+or, written without mutation, (and where `+` is mod $2^32$),
+
+```
+a'  = a + b ; d' = (d ⊕ a') <<< 16;
+c'  = c + d'; b' = (b ⊕ c') <<< 12;
+a'' = a' + b'; d'' = (d' ⊕ a') <<< 8;
+c'' = c' + d''; b'' = (c'' ⊕ b') <<< 7;
+```
+
+We lay each line as two rows.
+
+Each line has the form
+
+```
+x += z; y ^= x; y <<<= k
+```
+
+or without mutation,
+
+```
+x' = x + z; y' = (y ⊕ x') <<< k
```

Comment:
perhaps you want to have a mapping saying what variables will `x, y, z` correspond to. Example, if `x` is `a` then `z` is `b` but not `c` nor `d`

Reply:
I am not sure if in ChaCha it could look similar, but here's a graphical representation that I did for Salsa

<img width="903" alt="Captura de pantalla 2022-03-28 a las 19 27 53" src="https://user-images.githubusercontent.com/16045022/160453886-99fa9692-2dfa-4bcc-b66b-f8f3c6e0e370.png">
<img width="888" alt="Captura de pantalla 2022-03-28 a las 19 28 11" src="https://user-images.githubusercontent.com/16045022/160453917-5c9c4daa-dd03-4438-8b37-1f9e828e295d.png">
<img width="912" alt="Captura de pantalla 2022-03-28 a las 19 28 46" src="https://user-images.githubusercontent.com/16045022/160453994-e3365ea5-7e63-485d-957c-7fabbc54cfae.png">
<img width="877" alt="Captura de pantalla 2022-03-28 a las 19 28 56" src="https://user-images.githubusercontent.com/16045022/160454019-b0894b3f-7624-4c59-9541-00c47f67ae38.png">